hail/src/main/scala/is/hail/cxx/Emit.scala

```diff
@@ -797,6 +797,59 @@ class Emitter(fb: FunctionBuilder, nSpecialArgs: Int, ctx: SparkFunctionContext)
            """.stripMargin,
           resultRegion)
 
+      case ir.MakeNDArray(data, shape, rowMajor) =>
+        val elementPType = data.pType.asInstanceOf[PContainer].elementType
+        val datat = emit(data)
+        val shapet = emit(shape)
+        val rowMajort = emit(rowMajor)
+
+        val elemSize = elementPType.byteSize
+        val elemAlignment = elementPType.alignment
+        val flags = fb.variable("flags", "int", s"${rowMajort.v} ? 1 : 0")
+        val shapeVec = fb.variable("shapeVec", "std::vector<long>",
+          s"load_vector<long, true, 8, 8>(${shapet.v})")
```

Comment:
the parameters for the `load_vector` function should come from the `shape` type--i.e.

```
val shapeElt = coerce[PContainer](shape.pType).elementType
load_vector<${ typeToCxxType(shapeElt) }, ${ shapeElt.required }, ...>()
```

I'm actually a little surprised that tests are passing, I guess, since I'd expect the shape triplet to, in general, not have required elements.

Reply:
(the more I think about this, the weirder I feel about it. Is there something obvious that I'm missing that makes this correct? If not we should do some digging as to why the tests are passing here.)

Reply:
oh, I think I've got it. In general, if you're assuming that the shape is type `TArray(TInt64())`, the 8-byte alignment will mean that the elements are in the same location for all arrays up to size `4 * 8`. Consider [1L, 2L, 3L]:

```
non-required
03 00 00 00 00 00 00 00 01 00 00 00 00 00 00 00 02 00 ...
 --length--|missingness|-----data--->

required:
03 00 00 00 00 00 00 00 01 00 00 00 00 00 00 00 02 00 ...
 --length--|--padding--|-----data--->

```

Reply:
Hm ya I totally just rested on the assumption that they're typechecked to be arrays of longs, but didn't consider the alignment. Will used the type values.